src/Resources/RelationManager/EditRecord.php

```diff
@@ -99,6 +101,16 @@ public function switchRecord($manager, $recordKey)
         $this->callHook('afterFill');
     }
 
+    public function switchRecordInstance($record)
```

Comment:
And as mentioned I'm not sure what the best or most logical way to acquire the proper record/model is.
I'm sure that i am overlooking a more simple way to do this that's already supported.